packages/calypso-e2e/src/lib/pages/user-signup-page.ts

```diff
@@ -47,4 +57,39 @@ export class UserSignupPage {
 			this.page.click( selectors.submitButton ),
 		] );
 	}
+
+	/**
+	 * Signup form that is used by WordPress.com Connect (WPCC) endpoint.
+	 *
+	 * WPCC is a single sign-on service. For more information, please see
+	 * https://wordpress.com/support/wpcc-faq/.
+	 *
+	 * @param {string} email Email address of the new user.
+	 * @param {string} password Password of the new user.
+	 */
+	async signupWPCC( email: string, password: string ): Promise< void > {
+		const target = getTargetDeviceName();
+
+		// On mobile devices, the signup form is not shown by default.
+		if ( target === 'mobile' ) {
+			await this.page.click( selectors.createWPCOMAccountButton );
+		}
+
+		await this.page.fill( selectors.firstNameInput, 'E2E' );
+		await this.page.fill( selectors.lastNameInput, 'Testing' );
+		await this.page.fill( selectors.emailInput, email );
+		await this.page.fill( selectors.passwordInput, password );
+
+		if ( target === 'mobile' ) {
+			await Promise.all( [
+				this.page.waitForNavigation(),
+				this.page.click( selectors.submitButton ),
+			] );
+		} else {
+			await Promise.all( [
+				this.page.waitForNavigation(),
+				this.page.click( selectors.createWPCOMAccountButton ),
+			] );
+		}
```

Comment:
Do you think it would be better to make the submit button selector depend on the target device?

```js
const selectors = {
  // ...
  submitButton: target === 'mobile'
    ? 'button:text("Create a WordPress.com Account"):visible'
    : 'button[type="submit"]',
}
```

Reply:
Yes and no - in mobile, the user must first click on 'Create a WordPress.com account' button to reveal the form. The button with the same text then becomes a submit button.

However on desktop the 'Create a WordPress.com account' button is always visible and behaves as the submit button.

Here's an alternative solution I came up with while poking around in the DOM a bit more:

```
const selectors = {
   submitButton: button[type="submit"],
   createWPAccountButton: button:text("Create a WordPress.com Account"):visible
}
```

then, the relevant part of the `signupWPCC` method can simply call:

```
await Promise.all( [
	this.page.waitForNavigation(),
	this.page.click( selectors.submitButton ),
] );
```

Reply:
Fixed in c06160a60fea8e7e37c9965e404be4fc1c25e7bf.